src/components/projects/projectId/settings/embeddings/Embeddings.tsx

```diff
@@ -71,6 +71,14 @@ export default function Embeddings(props: { refetchEmbeddings: () => void }) {
     const orgId = useSelector(selectOrganizationId);
     useWebsocket(orgId, Application.REFINERY, CurrentPage.PROJECT_SETTINGS, handleWebsocketNotification, projectId, CurrentPageSubKey.EMBEDDINGS);
 
+    const openFilterAttributesModal = useCallback((embedding: Embedding) => () => {
+        embedding.onQdrant ? dispatch(setModalStates(ModalEnum.FILTERED_ATTRIBUTES, { embeddingId: embedding.id, open: true, attributeNames: prepareAttributeDataByNames(embedding.filterAttributes), showEditOption: showEditOption })) : null;
+    }, [showEditOption]);
+
+    const embeddingsFinal = useMemo(() => embeddings.map((embedding) => (
+        { ...embedding, onIconNotesClick: openFilterAttributesModal(embedding) }
+    )), [embeddings]);
```

Comment:
the function pointer isn't replaced when it's changed.

Easier to directly use the logic inside the Memo

```suggestion
    const embeddingsFinal = useMemo(() => embeddings.map((embedding) => (
        { ...embedding, onIconNotesClick: embedding.onQdrant ? dispatch(setModalStates(ModalEnum.FILTERED_ATTRIBUTES, { embeddingId: embedding.id, open: true, attributeNames: prepareAttributeDataByNames(embedding.filterAttributes), showEditOption: showEditOption }) }
    )), [embeddings, showEditOption /* or a s ref*/ ]);
```

not sure i got all the brackets right :D